io/net/vnic_dlpar.py

```diff
@@ -0,0 +1,215 @@
+        self.lpar_profile_name = self.params.get("lpar_profile_name", '*',
+                                                 default=None)
+        self.slot_num = self.params.get("slot_num", '*', default=None)
+        self.vios_name = self.params.get("vios_name", '*', default=None)
+        self.vios_id = self.params.get("vios_id", '*', default=None)
+        self.sriov_port = self.params.get("sriov_port", '*', default=None)
+        self.sriov_adapter_id = self.params.get("sriov_adapter_id", '*',
+                                                default=None)
+        self.bandwidth = self.params.get("bandwidth", '*', default=None)
+        self.login(self.hmc_ip, self.hmc_username, self.hmc_pwd)
+        self.run_command("uname -a")
+        cmd = 'lshwres -r io -m ' + self.server + \
+              ' --rsubtype slot --filter lpar_names=' + self.lpar + \
+              ' -F lpar_id'
+        self.lpar_id = self.run_command(cmd)[-1]
+        self.backing_devices = "backing_devices=sriov/%s/%s/%s/%s/%s"\
+                               % (self.vios_name, self.vios_id,
+                                  self.sriov_adapter_id, self.sriov_port,
+                                  self.bandwidth)
+
+    def login(self, ip, username, password):
+        '''
+        SSH Login method for remote server
+        '''
+        p = pxssh.pxssh()
+        # Work-around for old pxssh not having options= parameter
+        p.SSH_OPTS = p.SSH_OPTS + " -o 'StrictHostKeyChecking=no'"
+        p.SSH_OPTS = p.SSH_OPTS + " -o 'UserKnownHostsFile /dev/null' "
+        p.force_password = True
+
+        p.login(ip, username, password)
+        p.sendline()
+        p.prompt(timeout=60)
+        # Ubuntu likes to be "helpful" and alias grep to
+        # include color, which isn't helpful at all. So let's
+        # go back to absolutely no messing around with the shell
+        p.set_unique_prompt()
+        p.prompt(timeout=60)
+        self.pxssh = p
+
+    def run_command(self, command, timeout=300):
+        '''
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on hmc %s", command)
+        c = self.pxssh
+        c.sendline(command)
+        c.expect("\n")  # from us
+        c.expect(c.PROMPT, timeout=timeout)
+        output = c.before.splitlines()
+        c.sendline("echo $?")
+        c.prompt(timeout)
+        return output
+
+    def test(self):
+        '''
+        Start rsct services and executes vnic add and remove operations
+        '''
+        try:
+            process.run("startsrc -g rsct", shell=True, sudo=True)
+        except CmdError as details:
+            self.log.debug(str(details))
+            self.fail("Command startsrc -g rsct failed")
+
+        try:
+            process.run("startsrc -g rsct_rm", shell=True, sudo=True)
+        except CmdError as details:
+            self.log.debug(str(details))
+            self.fail("Command startsrc -g rsct_rm failed")
```

Comment:
These are doing the same operation for `rsct` and `rsct_rm`. Same code can be re-used.